qt/widgets/instrumentview/test/InstrumentWidget/InstrumentWidgetTest.h

```diff
@@ -299,6 +314,30 @@ class InstrumentWidgetTest : public CxxTest::TestSuite {
     draw_tab_save_actions("test_ws_d", 2);
   }
 
+  void test_peak_with_no_detector() {
+    for (const bool useLoadingThread : {true, false}) {
+      auto qtMock = makeQtDisplay();
+      auto glMock = makeGL();
+      auto displayMock = makeDisplay();
+      EXPECT_CALL(*displayMock, currentWidget()).Times(1).WillOnce(Return(qtMock.get()));
+
+      auto widget = constructWithUnwrappedSurface("test_ws", std::move(displayMock), qtMock.get(), glMock.get(), 23,
+                                                  useLoadingThread);
+      widget.setViewType("CYLINDRICAL_X");
+
+      auto createPeaksWs = Mantid::API::AlgorithmManager::Instance().create("CreatePeaksWorkspace");
+      createPeaksWs->setRethrows(true);
+      createPeaksWs->setProperty("InstrumentWorkspace", "test_ws");
+      createPeaksWs->setProperty("OutputWorkspace", "peaks");
+      createPeaksWs->execute();
+      auto ws = AnalysisDataService::Instance().retrieve("peaks");
+      auto peaksWs = std::dynamic_pointer_cast<IPeaksWorkspace>(ws);
+      auto peak = Mantid::DataObjects::Peak(peaksWs->getInstrument(), Mantid::Kernel::V3D(1, 1, 1));
+      peaksWs->addPeak(peak);
+      widget.overlay("peaks");
```

Comment:
up to you but I like to add `TS_ASSERT_THROWS_NOTHING` around the line where I'm testing no crash occurs just to make it explicit what the point of the test is. I do understand it's pretty redundant though :upside_down_face: